Opinionated fixes: update nuxt, use dayjs for formatting, npm start tailored to devs

asyncjake opened the pull request:
The title's a good one - the background information: I just spent most of my weekend trying to find some combination of a static site generator and a good starting point to work from, and I happened to like maraho on nuxt a lot. Since I was able to get it running with the timezone fix in PR#8, I figured I'd pass along these more opinionated updates in another PR as thanks for the awesome starting point. No offense taken if these aren't changes you like, but it's where I was able to hit the ground running on my site, so I figured there's no loss passing those changes up the chain in case they're helpful.

So yeah, thanks for maraho! Hope you're having a good weekend 😄

Commit:
Smallest possible fix for the timezone issue - solves site unable to start in timezones west of UTC

Commit:
Updating Next (verified working locally), adjusting date formatter to be more flexible with 8kb library, and tweaking 'npm start' to be the default dev command

cheetosysst commented:
I'll check it out when I have free time, within a week hopefully.

cheetosysst commented:
on first glance, I noticed the new date format you're using `DD MMMM YYYY`.
Personally I prefer `YYYY-MM-DD`. I wonder if you have ideas on how we could support both?

asyncjake commented:
Hey @cheetosysst :D I just tried to match the previous version, but yeah this could definitely handle both, do you think it makes sense to confugure that at build time?

If so, I looked into [next runtime config](https://nextjs.org/docs/pages/api-reference/next-config-js/runtime-configuration) and found the official docs recommend just [using `env` for it](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables), which is easy to set default values for and add some suggested date formats in the comments around it. That would just be two changes: one is a new default env file at maraho/.env like
```
# Format for blog page dates. Examples:
#   "ddd, DD MMMM YYYY HH:mm:ss ZZ" will give you "Mon, 02 January 2024 09:37:42 +0600"
#   "YYYY-MM-DD HH:mm" will give you "2023-11-25 21:37"
#   
# For more information, see https://day.js.org/docs/en/display/format
DAYJS_FORMAT="YYYY-MM-DD HH:mm"
```

And in `components/xml.js` we'd just update the format string to something like
```
const dayjsFmt = process.env.DAYJS_FORMAT || "YYYY-MM-DD HH:mm";
// alternately
if (!process.env.DAYJS_FORMAT)
    panicAndCry();
const dayjsFmt = process.env.DAYJS_FORMAT;
````

If you want that format to exist in the client after building the site, there's a fancy prefix (smth like NEXT_PUBLIC_VARGOESHERE) where all references get transpiled to the value of NEXT_PUBLIC_VARGOESHERE at build time. Open to ideas, .env seems to be a solid standard though

asyncjake commented:
Verbose mode on the comments is intentional since I'm thinking about this from a "random devs use this and need to be self-supporting" perspective, dunno if that makes sense in this case but I'm just sharing my process

cheetosysst commented:
```ts
const dayjsFmt = process.env.DAYJS_FORMAT || "YYYY-MM-DD HH:mm";
```
I like this one

Commit:
Custom date formats now configurable via env with a good default

asyncjake commented:
Forgot to follow up here: I just updated it to use env and added some dev-friendly comments for context, and verified via hot-reload and console logs that .env is loading correctly and the format is working

cheetosysst commented on package.json:
```diff
@@ -3,9 +3,9 @@
+    "start": "next dev",
     "dev": "next dev",
     "build": "next build",
-    "start": "next start",
```
Any particular reason we're changing this?

asyncjake replied:
Could be critical top secret reasons regarding government security, or because a random code monkey felt like `npm start` was shorter than `npm run dev`. Tbh I never even checked to see if the commands mapping to next commands was expected by next or vercel, and given this is set up this way by `next init` I bet this breaks something in the default deploy setup 😂 

So yeah I can back this out, lmk if there's anything else you'd want changed before merging and I'll get it all in one commit

Commit:
Reverting package script change because I forgot they were autogenerated and I don't know if they're expected
